openlibrary/plugins/openlibrary/js/ol.js

```diff
@@ -405,6 +405,14 @@ $(function(){
         $(this).closest('.arrow').toggleClass("up");
     }, 300, false));
 
+    $(document).on('click', function(e) {
+        var container = $("#widget-add");
```

Comment:
My read of this function is as follows:

For any click on the document...
if  user did not click on "#widget-add"....
and the element clicked on is NOT inside #widget-add
slideUp and removeClass.

So basically, you want to catch any clicks outside #widget-add, correct?

If so, you can do this is what stopPropagation is for - it lets you make sure clicks don't bubble upwards.

If I'm not mistaken this code does the same and is much more readable/performant:
```
$(document).on('click', function(e) {
            $(container).find('.dropdown').slideUp(25);
            $(container).find('.arrow').removeClass("up");
})
// Make sure clicks inside the widget-add don't bubble up to above click handler
$("#widget-add").on('click', function (e) { e.stopPropagation(); } );

```